boot/bootstate20.go

```diff
@@ -176,18 +181,20 @@ func (u20 *bootStateUpdate20) commit() error {
 		modeenvRewritten = true
```

Comment:
I'm confused, that's the point, the reseal code doesn't reseal for real if the boot chains haven't changed and we didn't include gadget info int those, so we need to worry only about the special case of unasserted stuff that's why my suggestion

Reply:
Ok, so I understand that the problem is that `resealKeyToModeenv()` should find that resealing is not necessary in this case by itself, and somewhat that is not the case. I'll see if I can find out the cause for that.

Reply:
I don't think it's likely an issue with resealKeyToModeenv, is just that a lot of our tests use unasserted kernels, so the flag becomes important and my suggestion is to redefine it based whether anything but gadget changed

Reply:
Ok, I've pushed changes following your suggestion. To me, the confusion comes from the flag applying only if there are unasserted kernels. Should not the behavior be the same for asserted ones? If, for instance, a new gadget is installed so modeenv changes but there are no bootchain assets changed by it. Why would we need to reseal the key in that case? What PCR register would change?

Reply:
I have the impression that you haven't looked how the flag works in detail, maybe because the name is not great. It is only used a hint, so in your scenario we don't reseal.

Reply:
the kernel is tracked in the boot chain as kernel-name/revision (see bootchain.go), this is well defined for a store asserted snap, for a local unasserted snap we don't have a unique revision so we opt to reseal more instead, this is also not  a bad compromise test wise. We could have used a hash for the kernel but that needed more changes (because is not something we easily have available in boot as is) everywhere and be more expensive in some cases. So we went for this.

Reply:
Thanks for the explanation!